rusty-machine/src/linalg/matrix/decomposition.rs

```diff
@@ -159,7 +160,7 @@ impl<T: Copy + Zero + Float> Matrix<T> {
     }
 }
 
-impl<T: Copy + Zero + One + Float + NumCast + Signed> Matrix<T> {
+impl<T: Any + Copy + Zero + One + Float + NumCast + Signed> Matrix<T> {
```

Comment:
I Copy, Zero, One, NumCast are all redundant here (included in Float, if that's num::Float).